pkg/archive/diff.go

```diff
@@ -40,12 +42,30 @@ func UnpackLayer(dest string, layer ArchiveReader) (size int64, err error) {
 		// Normalize name, for safety and for a simple is-root check
 		hdr.Name = filepath.Clean(hdr.Name)
 
-		if !strings.HasSuffix(hdr.Name, "/") {
+		// Windows does not support filenames with colons in them. Ignore
+		// these files. This is not a problem though (although it might
+		// appear that it is). Let's suppose a client is running docker pull.
+		// The daemon it points to is Windows. Would it make sense for the
+		// client to be doing a docker pull Ubuntu for example (which has files
+		// with colons in the name under /usr/share/man/man3)? No, absolutely
+		// not as it would really only make sense that they were pulling a
+		// Windows image. However, for development, it is necessary to be able
+		// to pull Linux images which are in the repository.
```

Comment:
So, let's turn this around... if its a windows image, how did the file get in there?  And if it is in there then perhaps we shouldn't silently ignore it we should error on the pull because we can't do what the user asked. 
If its a linux image then it seems like they should get an error because it won't work on a window daemon and, again, we can't do what the user asked so we need to tell them. Silently ignoring stuff is ok in some cases, but how would a user ever figure out why certain files are missing when they probably won't have access to the daemon logs?

Net: I'm wondering if we should error in these cases.